dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2022 HPMicro
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The HPMicro pin controller is a singleton node responsible for
+  controlling pin function selection and pin properties. For example, you can
+  use this node to route UART0 RX to pin PA10 and enable the pull-up resistor
+  on the pin. Remapping is also supported.
+
+  The node has the 'pinctrl' node label set in your SoC's devicetree,
+  so you can modify it like this:
+
+    &pinctrl {
+            /* your modifications go here */
+    };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+  &pinctrl {
+    pinmux_uart0: pinmux_uart0 {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTY, 7),IOC_TYPE_IOC, 0, 2)>,
+        <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTY, 6),IOC_TYPE_IOC, 0, 2)>;
+      };
+    };
+    pinmux_gpiob: pinmux_gpiob {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 18),IOC_TYPE_IOC, 0, 0)>,
```

Comment:
suggestion, since this seems unnecessarily verbose:

```
<HPMICRO_PINMUX(PORTB, 18, IOC, 0, 0)>,
```

(and use macro concatenation within HPMICRO_PINMUX)